graph-subscriptions-api/src/schema.rs

```diff
@@ -91,11 +93,81 @@ impl From<datasource::RequestTicket> for RequestTicketDto {
         Self {
             id,
             ticket_user: value.ticket_user,
-            ticket_signer: value.ticket_signer,
             ticket_name: value.ticket_name,
             total_query_count: value.total_query_count,
             queried_subgraphs_count: value.queried_subgraphs_count,
             last_query_timestamp: value.last_query_timestamp,
+            ticket_payload: value.ticket_payload.into(),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct TicketPayloadDto {
+    pub signer: String,
+    pub allowed_domains: Option<String>,
+    pub allowed_deployments: Option<String>,
+    pub allowed_subgraphs: Option<String>,
+}
+
+impl From<datasource::GatewaySubscriptionQueryResultTicketPayload> for TicketPayloadDto {
+    fn from(value: datasource::GatewaySubscriptionQueryResultTicketPayload) -> Self {
+        Self {
+            signer: value.signer,
+            allowed_domains: value.allowed_domains,
+            allowed_deployments: value.allowed_deployments,
+            allowed_subgraphs: value.allowed_subgraphs,
+        }
+    }
+}
+
+#[Object]
+impl TicketPayloadDto {
+    /// The ticket signer.
+    /// For most use-cases, this will match the `RequestTicket.ticket_user` value.
+    /// But if the signer is an authorized signer on the subscription, this value will be the signing wallet
+    pub async fn signer(&self) -> String {
+        self.signer.to_string()
+    }
+    /// A list of Subgraph Deployment Qm hashes the ticket is allowed to query
+    pub async fn allowed_deployments(&self) -> Vec<String> {
+        match &self.allowed_deployments {
+            Some(deployments) => deployments.split(",").map(|d| d.to_string()).collect(),
+            None => vec![],
+        }
```

Comment:
My suggestion isn't necessarily "better". However, an important note: split doesn't behave as most people expect ([docs](https://doc.rust-lang.org/std/primitive.str.html#method.split)). I've seen this cause bugs in both graph-node & gateway. I find `split_terminator` to usually be the right choice, so check you're uses of `split`.
```suggestion
        self.allowed_domains
            .as_deref()
            .unwrap_or_default()
            .split_terminator(',')
            .map(ToString::to_string)
            .collect()
```

Reply:
Got it. Thanks for the heads up!